lib/ruby-lint/constant_loader.rb

```diff
@@ -70,13 +77,37 @@ def run(ast)
     #
     def after_initialize
       @loaded = Set.new
+      @module_nesting = []
+      @registry ||= RubyLint.registry
+    end
+
+    def on_module(node)
+      name, _body = *node
+      cp = ConstantPath.new(name)
+
+      @module_nesting.push(cp.to_s)
+    end
+
+    def after_module(_node)
+      @module_nesting.pop
+    end
+
+    def on_class(node)
+      name, _parent, _body = *node
+      cp = ConstantPath.new(name)
+
+      @module_nesting.push(cp.to_s)
+    end
+
+    def after_class(_node)
+      @module_nesting.pop
     end
 
     ##
     # @param [RubyLint::Node] node
     #
     def on_const(node)
-      load_constant(ConstantPath.new(node).root_node[1])
+      try_load_constant(ConstantPath.new(node).root_node[1])
```

Comment:
Since this method tries to load constants while dealing with nesting I'd rather have it named something like `load_nested_constant`.